Add unrated puzzles option

CloudyDino opened the pull request:
# Unrated Puzzles
Close https://github.com/lichess-org/mobile/issues/1120
Close https://github.com/lichess-org/mobile/issues/297

## Changes
- Adds a switch to the puzzle settings screen to let users play unrated puzzles that don't affect their rating.

https://github.com/user-attachments/assets/0c7f35d1-528b-4b7e-929f-1a8d1a2566a8

- Logged out users don't see the rating switch:

https://github.com/user-attachments/assets/42ecd261-d12f-48f4-a6ef-9704b4f97f3f

## Testing
Tested using a proxy to spy on what app sends to API.

https://github.com/user-attachments/assets/90a00036-56e5-4273-9f2e-275e8c3c9f7c

1. First puzzle is correctly `win: false, rated: true`
```json
{
	"solutions": [{
		"id": "24pDF",
		"win": false,
		"rated": true
	}]
}
```
2. Second puzzle is correctly `win: true, rated: true`
```json
{
	"solutions": [{
		"id": "7nI7d",
		"win": true,
		"rated": true
	}]
}
```
3. Third puzzle is correctly `win: true, rated: false`
```json
{
	"solutions": [{
		"id": "6YIPJ",
		"win": false,
		"rated": false
	}]
}
```
6.  Rest of the puzzles are played offline and are correctly rated once I turn the internet back on and the batch is submitted.
```json
{
	"solutions": [{
		"id": "8Cm4H",
		"win": true,
		"rated": false
	}, {
		"id": "0haqU",
		"win": true,
		"rated": true
	}, {
		"id": "5T8Mz",
		"win": true,
		"rated": true
	}, {
		"id": "2106v",
		"win": false,
		"rated": true
	}, {
		"id": "2ZCbv",
		"win": false,
		"rated": false
	}, {
		"id": "9h38Q",
		"win": true,
		"rated": false
	}]
}
```

veloce commented:
Thanks for this!

How does it work with batches? Puzzles work offline too. If you solve puzzle offline, the solution is saved until network is up again. Then batches of puzzle are solved at once.

CloudyDino commented:
@veloce Good catch! I updated the testing section above with a video that shows me turning on/off my wifi and turning on/off the rating. I then checked that once the internet was back, the batch puzzle submissions were all correctly rated or not rated. I think this should be ready for a final review

veloce commented:
I've rebased your branch to fix the CI @CloudyDino

CloudyDino commented:
@veloce 
Removed the offending line `await tester.pumpAndSettle()`.
Renamed `rated` to `isRatedPreference`.
Once this is merged in, I'll base the hints branch off of this for the tests over there

veloce reviewed (approved):
LGTM. Great job, thanks!
